prusti-encoder/src/encoders/mir_pure_function.rs

```diff
@@ -1,47 +1,62 @@
-use prusti_rustc_interface::{middle::{mir, ty}, span::def_id::DefId};
+use prusti_rustc_interface::{
+    middle::ty::{self, GenericArgs},
+    span::def_id::DefId,
+};
 
 use task_encoder::{TaskEncoder, TaskEncoderDependencies};
-use vir::{Reify, FunctionIdent, UnknownArity, CallableIdent};
 
-use crate::encoders::{
-    MirPureEnc, MirPureEncTask, mir_pure::PureKind, MirSpecEnc, MirLocalDefEnc,
+use crate::encoder_traits::pure_function_enc::{
+    PureFunctionEnc, MirFunctionEncOutput, MirFunctionEncOutputRef,
 };
 
-use super::PredicateEncOutputRef;
 pub struct MirFunctionEnc;
 
 #[derive(Clone, Debug)]
 pub enum MirFunctionEncError {
     Unsupported,
 }
 
-#[derive(Clone, Debug)]
-pub struct MirFunctionEncOutputRef<'vir> {
-    pub function_ref: FunctionIdent<'vir, UnknownArity<'vir>>,
-}
-impl<'vir> task_encoder::OutputRefAny for MirFunctionEncOutputRef<'vir> {}
+impl PureFunctionEnc for MirFunctionEnc {
+    fn get_substs<'tcx>(
+        vcx: &vir::VirCtxt<'tcx>,
+        def_id: &Self::TaskKey<'tcx>,
+    ) -> &'tcx GenericArgs<'tcx> {
+        GenericArgs::identity_for_item(vcx.tcx(), *def_id)
+    }
 
-#[derive(Clone, Debug)]
-pub struct MirFunctionEncOutput<'vir> {
-    pub function: vir::Function<'vir>,
+    fn mk_function_ident<'vir, 'tcx>(
+        vcx: &'vir vir::VirCtxt<'tcx>,
+        def_id: &Self::TaskKey<'tcx>,
+    ) -> vir::ViperIdent<'vir> {
+        vir::vir_format_identifier!(vcx, "f_{}_CALLER", vcx.tcx().item_name(*def_id))
```

Comment:
I guess there is no reason for this to say "CALLER" anymore? IIUC that bit was there to separate what is being encoded from what is calling this.

Reply:
Removed it!